docs/source/concepts/architectural-model.rst

```diff
@@ -0,0 +1,284 @@
+Establishing such an order at all endpoints incurs synchronization, which hinders scalability.
+Different endpoints are therefore allowed to output events in different orders if these events do not causally depend on each other from the party's perspective.
+This section defines :ref:`causality <ledger-causality-party>` and illustrates the allowed and disallowed reorderings of commits by :ref:`examples <ledger-causality-examples>`.
+
+.. _ledger-causality-party:
+
+Causality
+=========
+
+Causality is defined on the commits in a party's projection of the virtual shared ledger.
+
+Definition »synchronizes with«
+  An action `act`\ :sub:`1` **synchronizes with** an action `act`\ :sub:`2` for party `A` if one of the following holds:
+
+  * **Create-use:**
+    `act`\ :sub:`1` creates the contract `c` and `act`\ :sub:`2` uses the contract, i.e., `act`\ :sub:`2` is a **Fetch** or an **Exercise** on `c` and `A` is a stakeholder of `c` and an informee of `act`\ :sub:`2`.
+ 
+  * **Use-use:**
+    `act`\ :sub:`1` and `act`\ :sub:`2` both use a contract `c` and `A` is a stakeholder of `c` and an informee of both actions.
+
+  * **Contract keys**:
+    `act`\ :sub:`1` and `act`\ :sub:`2` both are actions on a key `k` and `A` is a maintainer of `k`.
+
+Definition »causal order«
+  A commit `C`\ :sub:`2` **causally depends on** another commit `C`\ :sub:`1` for party `A` if all of the following holds:
+  * `C`\ :sub:`1` precedes `C`\ :sub:`2` on the virtual shared ledger.
+  * They contain actions `act`\ :sub:`2` and `act`\ :sub:`1` such that `act`\ :sub:`1` synchronizes with `act`\ :sub:`2` for party `A`.
+  
+  A commit `C`\ :sub:`1` is **causally ordered** before `C`\ :sub:`2` for `A` if `C`\ :sub:`2` causally depends on `C`\ :sub:`1` for `A` or if there is another commit `C'` in `A`\ 's projection such that `C`\ :sub:`1` is causally ordered before `C'` for `A` and `C'` is causally ordered before `C`\ :sub:`2` for `A`.
+
+.. note::
+   The synchronizes with relationship is not symmetric.
+   If `act`\ :sub:`1` uses a contract and `act`\ :sub:`2` creates it, then `act`\ :sub:`1` does not synchronize with `act` :sub:`2`.
+   This is in line with how the concept is used in programming languages: writes to a memory location synchronize with reads, but reads do not synchronize with writes.
+   This asymmetry does not matter for the causal ordering in a consistent ledger as **Create**\ s precede usages on such a ledger.
+  
+.. _ledger-api-guarantees:
+  
+Ledger API ordering guarantees
+==============================
+
+Subject to the trust assumptions of the deployed ledger,
+if a party `A` is hosted on an endpoint,
+the endpoint's transaction service output for `A` satisfy the following guarantees:
+
+#. The transaction trees on the transaction tree stream are precisely those that are in the endpoint's view for `A`.
+   This excludes ``LeftViewEvent``\ s and ``EnteredViewEvent``\ s because they are not part of the virtual shared ledger.
```

Comment:
Ah, so that's actually interesting. By calling this out, you distinguish between the guarantees that the VSL gives you versus the guarantees that the VIEW on the VSL gives you.